benchmarking-cli/src/main.rs

```diff
@@ -0,0 +1,157 @@
+    // The CLI should take a path to the example directory as input, execute/prove the example, and output the following metrics:
+
+    // Cold compilation time (i.e., when the example is compiled with a new assembler).
+    // Hot compilation time (i.e., when the example is complied with an assembler which was already used to compile the same code).
+    // Execution time.
+    // Proving time.
+    // Verification time.
+
+    println!("============================================================");
+    println!("Benchmarking Miden examples");
+    println!("============================================================");
+
+    let args = Cli::parse();
+
+    // let's read the program
+    let program_string = fs::read_to_string(format!("../examples/{}.masm", &args.example))?;
+
+    // let's read the input fils
+    let input_string = fs::read_to_string(format!("../examples/{}.inputs", &args.example))
+        .map_err(|err| format!("Failed to open input file `{}` - {}", &args.example, err))?;
+    let inputs_des: InputFile = serde_json::from_str(&input_string)
+        .map_err(|err| format!("Failed to deserialize input data - {}", err))
+        .unwrap();
+
+    let stack_init = get_stack_init(&inputs_des);
+    let advice_tape = get_advice_tape(&inputs_des);
+
+    let inputs = get_program_inputs(&stack_init, &advice_tape);
+
+    // Compilation time
+    let now = Instant::now();
+    let assembler = Assembler::new();
+    let program = assembler
+        .compile(&program_string)
+        .expect("Could not compile source");
+
+    println! {"Compilation Time (cold): {} ms", now.elapsed().as_millis()}
+
+    let now = Instant::now();
+    let _program2 = assembler
+        .compile(&program_string)
+        .expect("Could not compile source");
+
+    println! {"Compilation Time (hot): {} ms", now.elapsed().as_millis()}
+
+    // Execution time
+    let now = Instant::now();
+    let _trace = miden::execute(&program, &inputs)
+        .map_err(|err| {
+            format!(
+                "Failed to generate exection trace = {:?}, and advice_tape = {:?}",
+                err, advice_tape
+            )
+        })
+        .unwrap();
+
+    println! {"Execution Time: {} ms", now.elapsed().as_millis()}
+
+    // Proving time
+    let proof_options = if args.security == "high" {
+        ProofOptions::with_128_bit_security()
+    } else {
+        ProofOptions::with_96_bit_security()
+    };
+
+    let now = Instant::now();
+    let (output, proof) = miden::prove(&program, &inputs, &proof_options).expect("results");
+
+    println! {"Proving Time: {} ms", now.elapsed().as_millis()}
+
+    // Verification time
+    let program_input_u64 = inputs
+        .stack_init()
+        .iter()
+        .map(|x| x.inner())
+        .collect::<Vec<u64>>();
+
+    let now = Instant::now();
+    let _result = miden::verify(program.hash(), &program_input_u64, &output, proof)
+        .map_err(|err| format!("Program failed verification! - {}", err));
```

Comment:
If we're not using the result, we should be able to get rid of the `let _result =`, right?

Reply:
Actually I need that, `verify` might return an error. So if I remove the `let _result =` then the compiler complains about 

```
unused `Result` that must be used
`#[warn(unused_must_use)]` on by default
this `Result` may be an `Err` variant, which should be handled
```

And if I add an `unwrap()` at the end, it comiles but returns
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: "Program failed verification! - VerifierError(InconsistentOodConstraintEvaluations)"', src/main.rs:151:75
```

for the verifiaction step.